tests/transition.md

```diff
@@ -277,3 +281,381 @@ All test cases described in this document are beginning in a pre-Merge world and
+  * EL should pull `B: INV_TB <- P1 <- P2` from the network
+  * poll `forkchoiceUpdated(B.P2)`
+    * EL returns `{status: INVALID, latestValidHash: 0x00.00}`
+    * EL's head points to `A.P2`
+  
+  </details>
+
+* [ ] Re-org and sync with the chain having invalid *transition* block
+  <details>
+  <summary>Click for details &#9662;</summary>
+
+  * `A: Genesis <- ... <- TB <- P1 <- P2`, `B: Genesis <- ... <- TB <- INV_P1 <- P2`, `B.TB` is a valid terminal block, `B.INV_P1` is an invalid payload
+  * EL starts with `A.P2` as the head
+  * `newPayload(B.P2) + forkchoiceUpdated(B.P2)`
+    * EL returns `{status: SYNCING}`
+    * EL's head points to `A.P2`
+  * poll `forkchoiceUpdated(B.P2)`
+    * EL returns `{status: INVALID, latestValidHash: 0x00.00}`
+    * EL's head points to `A.P2`
+  
+  </details>
+
+* [ ] Re-org and sync with the chain having invalid *post-transition* block
+  <details>
+  <summary>Click for details &#9662;</summary>
+
+  * `A: Genesis <- ... <- TB <- P1 <- P2 <- P3`, `B: Genesis <- ... <- TB <- P1 <- INV_P2 <- P3`, `B.TB` is a valid terminal block, `B.INV_P2` is an invalid payload
+  * EL starts with `A.P2` as the head
+  * `newPayload(B.P3) + forkchoiceUpdated(B.P3)`
+    * EL returns `{status: SYNCING}`
+    * EL's head points to `A.P3`
+  * poll `forkchoiceUpdated(B.P3)`
+    * EL returns `{status: INVALID, latestValidHash: B.P1.blockHash}`
+    * EL's head points to `A.P3` (in this case `fcU` won't be applied at all because it does point to a block on an invalid chain, thus, no update of the forkchoice state should happen; normally, a CL will send `fcU(B.P1)` when get `lvh = B.P1.blockHash` and switch canonical chain to `B` with `B.P1` as the head)
+  
+  </details>
+
+* [ ] Stop processing gossiped PoW blocks
+  <details>
+  <summary>Click for details &#9662;</summary>
+
+  * PoW/Clique miner builds a chain and advertises it via gossip; the chain goes beyond a terminal PoW block
+  * EL is connected to the miner and does not process blocks after the terminal one, i.e. the head points to the terminal block
+  
+  </details>
+
+* [ ] Stop processing synced PoW blocks
+  <details>
+  <summary>Click for details &#9662;</summary>
+
+  * PoW client starts with `Genesis <- ... <- TB <- B1 <- B2 <- ... <- Bn` chain, where `TB` is a valid terminal block
+  * EL connects to the PoW client and syncs with the advertised chain
+    * EL's head points to `TB`
+  
+  </details>
+
+* [ ] Terminal blocks are gossiped
+  <details>
+  <summary>Click for details &#9662;</summary>
+
+  * Consider `PoW <-> EL1 <-> EL2` as a PoW client connected to EL client `EL1`, which in its turn connected to EL client `EL2`. There is no direct connection between `EL2` and `PoW` clients.
+  * `PoW` client gossips a terminal block `TB`
+    * Head of `EL2` client points to `TB`
+  * `PoW` client gossips a descendant of `TB`
+    * Head of `EL2` client points to `TB`
+  
+  </details>
+
+### CL client tests
+
+* [ ] Transition on a valid chain
```

Comment:
I think this can be covered with https://github.com/ethereum/hive/pull/552.
The main difference with the description here is that `Genesis==TB`, but we do check `safe` and `finalized` appropriately.